server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/utils/ErrorResponderTest.java

```diff
@@ -0,0 +1,112 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.utils;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Objects;
+
+import org.junit.Test;
+
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+
+import spark.HaltException;
+
+public class ErrorResponderTest {
+    @Test
+    public void haltErrorShouldThrowWhenNoStatusCode() throws Exception {
+        assertThatThrownBy(() -> ErrorResponder.builder()
+            .haltError())
+            .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    public void haltErrorShouldReturnBodyWithStatusCodeWhenSetting() throws Exception {
+        assertThatThrownBy(() -> ErrorResponder.builder()
+                .statusCode(400)
+                .haltError())
+            .isInstanceOf(HaltException.class)
+            .matches(e -> hasStatus(e, 400))
+            .matches(e -> bodyHasErrorDetail(e, new ErrorResponder.ErrorDetail(400, "", "", "")));
```

Comment:
I think **type** and **message** should be compulsory...

Reply:
Here **cause** should be and empty optional (be sure to have loaded Java 8 jackson bindings...)